### ISSUE TITLE
Calling MySQL functions on parameter values

### ISSUE THREAD

**Issue by TehShrike** [TehShrike/sql-concat#2]
Probably something like this:

``` js
q.where('some_column', q.fn('DATE', someDateTime)) // => WHERE some_column = DATE(?)
```

What should the syntax be for something like `DATE_ADD`, where the first argument would be the parameter value, and the second argument is some other arbitrary string?

What should the JavaScript look like that would produce `WHERE some_column = DATE_ADD(?, INTERVAL 1 DAY)`

**Comment by TehShrike** [comment]
I think the best/most flexible solution is probably to implemented tagged template literals (#8) and then change the where/having/join methods to allow passing in other query objects, instead of just regular values.

**Comment by TehShrike** [comment]
Okay, so, it shouldn't be an actual query object that gets passed in as the second argument to `where` or whatever - that would only make sense if it was a subquery.

That argument should instead be an object with `str` and `params` properties, that should make it easy to slot in.
